src/core_plugins/metrics/public/components/lib/replace_vars.js

```diff
@@ -2,9 +2,29 @@ import _ from 'lodash';
 import handlebars from 'handlebars/dist/handlebars';
 export default function replaceVars(str, args = {}, vars = {}) {
   try {
-    const template = handlebars.compile(str);
-    return template(_.assign({}, vars, { args }));
+    const template = handlebars.compile(str, { strict: true });
+
+    const string = template(_.assign({}, vars, { args }));
+
+    return string;
   } catch (e) {
-    return str;
+    // Unknown variable
+    if (e.message.indexOf('not defined in') !== -1) {
+      const badVar = e.message.split(/"/)[1];
+      e.error = {
+        caused_by: {
+          reason: `The variable "${badVar}" that you used is not avaliable in this data metric`,
```

Comment:
I would change the wording on this to read
```
`{{${badVar}}} is an unknown variable.`
```
Short and sweet :D

Reply:
fixed :)